src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -645,6 +648,31 @@ impl LeaderBlockCommitOp {
         self.check_single_burn_output()
     }
 
+    fn calculate_intended_height(
+        parent_modulus: u8,
+        actual_block_height: u64,
+    ) -> Result<u64, op_error> {
+        let intended_modulus = (parent_modulus as u64 + 1) % BURN_BLOCK_MINED_AT_MODULUS;
+        let actual_modulus = actual_block_height % BURN_BLOCK_MINED_AT_MODULUS;
+        if actual_modulus != intended_modulus {
```

Comment:
What happens if the block-commit is late by exactly `BURN_BLOCK_MINED_AT_MODULUS - 1` blocks, and `actual_block_height` is evenly divisible by `BURN_BLOCK_MINED_AT_MODULUS`?  This should never happen because the window size is smaller than `BURN_BLOCK_MINED_AT_MODULUS`, right?

Reply:
In this case, this check would fail to detect that this was a "missed" block commit, and instead it would try to process it as if it were a block commit in the current sortition (which would fail the block commit check if the commit outputs were invalid, and otherwise may be selected but could not produce a valid block).

This is possible, _but_ it's okay to "penalize" the miner in this case, because the window size has already passed -- trying to link commits that old won't improve the miner's score in the window.